.eslintrc.js

```diff
@@ -1,11 +1,80 @@
 module.exports = {
   parser: "@typescript-eslint/parser",
+  parserOptions: {
+    tsconfigRootDir: __dirname,
+    project: ["./tsconfig.json"],
+  },
   extends: [
+    "plugin:jest/recommended",
     "plugin:jest-dom/recommended",
+    "plugin:react/recommended",
     "plugin:react-hooks/recommended",
+    "plugin:react/recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    // "plugin:xstate/all", cos-ui // https://issues.redhat.com/browse/MGDOBR-510
+    // "plugin:storybook/recommended", app-services-ui-components
+    "eslint:recommended",
     "prettier",
   ],
-  plugins: ["@typescript-eslint", "react-hooks", "jest-dom", "testing-library"],
+  plugins: [
+    "@typescript-eslint",
+    "eslint-plugin-react-hooks",
+    "jest",
+    "jest-dom",
+    "react",
+    "react-hooks",
+    "testing-library",
+    //"xstate", cos-ui https://issues.redhat.com/browse/MGDOBR-510
+  ],
+
+  // kafka-ui
+  rules: {
+    "@typescript-eslint/no-unused-vars": "off",
```

Comment:
Weeeeell no unused vars. Always a pleasure to discuss it. :slightly_smiling_face: 

I prefer not having errors on unused vars (and imports) while developing because it can be a pain while you are trying solutions to always comment stuff before saving and rebuilding. At the same time it's nice to not leave unused stuff when committing code.

So turning this off on es lint would probably imply also turning off the `noUnusedLocals` from ts config. Otherwise we have a mix, compilation errors but no linting errors.

A compromise is to have `noUnusedLocals` turned off and keep `no-unused-vars` but only as a warning. So you are free to try things during development and then you have the chance to remove unused stuff to make linting warnings go away before submitting a PR. So with this approach we are trusting devs to remember to get rid of the warnings.

The alternative is to keep both eslint and tsconfig screaming on unused vars.

Reply:
I am happy to turn it on.